path/basename.ts

```diff
@@ -11,8 +12,12 @@ import { posixBasename, windowsBasename } from "./_basename.ts";
  * @param path - path to extract the name from.
  * @param [suffix] - suffix to remove from extracted name.
  */
-export function basename(path: string, suffix = ""): string {
-  return isWindows
+export function basename(
+  path: string,
+  suffix = "",
+  options?: PathOptions,
```

Comment:
Ideally this would be a single argument like
```ts
options?: {
  suffix?: string
  os?: OSType
}
```
but we don't live in an ideal world. Not sure what to do about this.